### PR TITLE
Added migration support for add_library and add_link_options CMake syntaxes

### PR DESCRIPTION
**Pull request by TejaX-Alaghari**
This PR adds migration support for add_library and add_link_options CMake syntaxes

### CODE REVIEW

**Review by tomflinda — CHANGES_REQUESTED** [review]
Pls rebase your PR to fix conflicts

File: clang/lib/DPCT/PatternRewriter.cpp [hunk 1]
```diff
@@ -273,9 +273,11 @@ static int parseCodeElement(const MatchPattern &Suffix,
     }
 
     const auto Character = Input[Index];
+
     if(Suffix.size() == 0 && Character =='"') {
       return Index;
     }
+
```
Comment (tomflinda) [inline clang/lib/DPCT/PatternRewriter.cpp]: The fixing PR https://github.com/oneapi-src/SYCLomatic/pull/1813 has been merged, pls rebase your PR to latest SYCLomatic branch.

File: clang/test/dpct/cmake_migration/case_037/expected.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+add_link_options()
+add_link_options()
+add_link_options()
+add_link_options( )
+add_link_options()
+
+add_link_options()
+add_link_options()
+add_link_options("")
+add_link_options("--no-ptxas-options")
```
Comment (zhiweij1) [inline clang/test/dpct/cmake_migration/case_037/expected.txt]: Is "--no-ptxas-options" an nvcc option? If so, can we remove it?

**Review by zhiweij1 — APPROVED** [review]

**Review by tomflinda — APPROVED** [review]
LGTM

**Comment by zhimingwang36** [comment]
Please solve the conflict.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**